server/src/main/java/io/crate/expression/operator/EqOperator.java

```diff
@@ -171,22 +155,12 @@ public static Query termsQuery(String column, DataType<?> type, Collection<?> va
             }
             return new TermInSetQuery(column, bytesRefs);
         }
-        return switch (type.id()) {
-            case StringType.ID -> new TermInSetQuery(column, nonNullValues.stream().map(BytesRefs::toBytesRef).toList());
-            case IntegerType.ID -> IntPoint.newSetQuery(column, (List<Integer>) nonNullValues);
-            case LongType.ID -> LongPoint.newSetQuery(column, (List<Long>) nonNullValues);
-            case FloatType.ID -> FloatPoint.newSetQuery(column, (List<Float>) nonNullValues);
-            case DoubleType.ID -> DoublePoint.newSetQuery(column, (List<Double>) nonNullValues);
-            case IpType.ID -> InetAddressPoint.newSetQuery(
-                column,
-                nonNullValues.stream().map(x -> InetAddresses.forString((String) x)).toArray(InetAddress[]::new)
-            );
-            case BitStringType.ID -> new TermInSetQuery(
-                column,
-                nonNullValues.stream().map(x -> new BytesRef(((BitString) x).bitSet().toByteArray())).toList()
-            );
-            default -> booleanShould(column, type, nonNullValues, hasDocValues, indexType);
-        };
+        StorageSupport<?> storageSupport = type.storageSupport();
+        EqQuery<?> eqQuery = storageSupport == null ? null : storageSupport.eqQuery();
+        if (eqQuery == null) {
+            return booleanShould(column, type, nonNullValues, hasDocValues, indexType);
+        }
```

Comment:
Afaik `StorageSupport.eqQuery()` can never be null.
```suggestion
        StorageSupport<?> storageSupport = type.storageSupport();
        if (storageSupport == null) {
            return booleanShould(column, type, nonNullValues, hasDocValues, indexType);
        }
        EqQuery<?> eqQuery = storageSupport.eqQuery();
```

Reply:
Thanks for reviewing. I had another look - GeoPoint, GeoShape, and ObjectTypes set it as nulls.